vue2-vuetify/src/complex/components/AdditionalProperties.vue

```diff
@@ -0,0 +1,463 @@
+      ) {
+        propSchema = control.value.schema.additionalProperties;
+      }
+
+      if (!propSchema && propValue !== undefined) {
+        // can't find the propertySchema so use the schema based on the value
+        // this covers case where the data in invalid according to the schema
+        propSchema = Generate.jsonSchema(
+          { prop: propValue },
+          {
+            additionalProperties: false,
+            required: () => false,
+          }
+        ).properties?.prop;
+      }
+
+      if (propSchema) {
+        if (propSchema.type === 'object' || propSchema.type === 'array') {
+          propUiSchema = Generate.uiSchema(propSchema, 'Group');
+          (propUiSchema as GroupLayout).label =
+            propSchema.title ?? startCase(propName);
+        } else {
+          propUiSchema = createControlElement(
+            control.value.path + '/' + encode(propName)
+          );
+        }
+      }
+
+      return {
+        propertyName: propName,
+        path: composePaths(control.value.path, propName),
+        schema: propSchema,
+        uischema: propUiSchema,
+      };
+    };
+
+    const appliedOptions = useControlAppliedOptions(props.input);
+    const additionalPropertyItems = ref<AdditionalPropertyType[]>([]);
+
+    additionalKeys.forEach((propName) => {
+      const additionaProperty = toAdditionalPropertyType(
+        propName,
+        control.value.data[propName]
+      );
+      additionalPropertyItems.value.push(additionaProperty);
+    });
+
+    const styles = useStyles(control.value.uischema);
+    const newPropertyName = ref<string | null>('');
+    const ajv = useAjv();
+
+    let propertyNameSchema: JsonSchema7 | undefined = undefined;
+    let propertyNameValidator: ValidateFunction<unknown> | undefined =
+      undefined;
+
+    // TODO: create issue against jsonforms to add propertyNames into the JsonSchema interface
+    // propertyNames exist in draft-6 but not defined in the JsonSchema
+    if (typeof (control.value.schema as any).propertyNames === 'object') {
+      propertyNameSchema = (control.value.schema as any).propertyNames;
+    }
+
+    if (
+      typeof control.value.schema.additionalProperties !== 'object' &&
+      typeof control.value.schema.patternProperties === 'object'
+    ) {
+      const matchPatternPropeertiesKeys: JsonSchema7 = {
+        type: 'string',
+        pattern: Object.keys(control.value.schema.patternProperties).join('|'),
+      };
+
+      propertyNameSchema = propertyNameSchema
+        ? { allOf: [propertyNameSchema, matchPatternPropeertiesKeys] }
+        : matchPatternPropeertiesKeys;
+    }
+
+    if (propertyNameSchema) {
+      propertyNameValidator = reuseAjvForSchema(
+        ajv,
+        propertyNameSchema
+      ).compile(propertyNameSchema);
+    }
+
+    const vuetifyProps = (path: string) => {
+      const props = get(appliedOptions.value?.vuetify, path);
+
+      return props && isPlainObject(props) ? props : {};
```

Comment:
Why is the `isPlainObject` check needed?

Reply:
this matched the vuetifyProps that we have currently in the composition file - there we also check the isPlainObject which I guess could be removed but it was checking that to guarantee that this is an object and not a simple data like String or number for example - e.g. with incorrect JSON files.